### DISCUSSION TITLE
Zipping multiple files into one

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6223] [Q&A]
I have a need to read PDF files from a directory and then combine them all into one zipped file before sending to the final destination. I've seen some post where MC can zip a file but is there a way to zip multiple files and make one zip file in one read?

**Comment by pacmano1** [comment]
Maybe start with https://github.com/pacmano1/Mirth-Snippets/blob/main/create_zip_with_wildcards.js

Not a full answer there but not entirely sure what you need to do.

**Comment by JoeFox82** [comment]
You can use the sevenzip java binding lib: https://sevenzipjbind.sourceforge.net/compression_snippets.html#update-add-remove-items

Or take a look at this: https://forums.mirthproject.io/forum/mirth-connect/support/16298-code-template-automatic-mirth-backup the script uses zip for adding multiple files for backup.

**Reply by pacmano1** [reply]
Both of those need extra jars.   Not a criticism there of the suggestion of course.

**Comment by rjkroll** [comment]
I use the following in a Destination transformer step for one of our interfaces that produces some files in a folder, then ZIPs then up.   This code also uses some extra stuff I am leaving in that I pass in to name the file that maybe you would find helpful.

```
importPackage(Packages.net.lingala.zip4j);
var path = String($('file_path'));
var outpath = "D:/Zips";

var files = new java.io.File(path).listFiles();
var results = new java.util.ArrayList();

var zipFile = new Packages.net.lingala.zip4j.core.ZipFile(outpath + "/" + String($('RequestID')) + "-" + String($('MRN')) + ".zip");

//If this pathname does not denote a directory, then listFiles() returns null. 
files.forEach(file =>{addFileToZip(results,path,file)});

var parameters = new Packages.net.lingala.zip4j.model.ZipParameters();
  parameters.setCompressionMethod(8);
  parameters.setCompressionLevel(9);
  
zipFile.addFiles(results, parameters);

function addFileToZip(acc,filePath,file){
	var fileName = file.getName();
	var fullFilePath = filePath + "/" +fileName
	if (file.isFile()) {
		acc.add(file);
		logger.info(fullFilePath);
	}
}
```
